app/src/main/java/at/roteskreuz/stopcorona/model/repositories/ReportingRepository.kt

```diff
@@ -283,6 +238,135 @@ class ReportingRepositoryImpl(
         }
     }
 
+    private suspend fun prepareListOfTemporaryExposureKeysForRedReporting(
+        temporaryExposureKeysFromSDK: List<TemporaryExposureKey>,
+        thresholdTime: Int
+    ): List<TemporaryExposureKeysWrapper> {
+        val temporaryExposureKeysList = mutableListOf<TemporaryExposureKeysWrapper>()
+
+        val sentYellowTemporaryExposureKeys =
+            infectionMessengerRepository.getSentTemporaryExposureKeysByMessageType(
+                MessageType.InfectionLevel.Yellow
+            )
+
+        // In case a yellow reporting happened in the past we report the keys reported yellow and
+        // the newer keys from Exposure SDK.
+        if (sentYellowTemporaryExposureKeys.isNotEmpty()) {
```

Comment:
I would prefer `sentYellowTemporaryExposureKeys.isEmpty()` and move the else block up

Reply:
I would keep the `sentYellowTemporaryExposureKeys.isNotEmpty()` as the first case because has a more complex logic and in this way gets more attention, the `isEmpty()` branch is the straightforward case.